### ISSUE TITLE
Endless complaints from users - Views not registering

### ISSUE THREAD

**Issue by para2022** [WWBN/AVideo#8281]
I had no problem with views on my system until that big update. Now they do not register. Sometimes they do most of the time they don't.

1. Affects firefox now as well as brave.
2.  If they register it is after 30-60secs (Big problem if users click off the vid before then)
3. I have changed php version, it makes no difference.
4. No errors in logs.
5. Users keep complaining all the time.
6. It is seriously affecting my sites views, they have dropped but active users are still the same as before.
7. I have not edited files, i have not touched the system other than updates.

Of course users are going to get angry if their views are being affected.

Thanks

**Comment by DanielnetoDotCom** [comment]
https://github.com/WWBN/AVideo/wiki/How-Video-Count-Works#checking-the-functionality

**Comment by para2022** [comment]
Yesterday i have had one my biggest channels of 1000+ videos whos been with me for 2 years leave. I contacted him to ask why as i was curious. He said views are a joke and he is being robbed of views so why bother. I am mad that you have ignored this and let it get to this stage. It takes exactly 30secs in brave browser for a view to register, if you leave before then the view doesnt register. He said whenever he checked his videos to see if they worked his view never registered. Because he was probs not staying on the video for 30 seconds so he had no idea, because this is a new problem. I told you 2 weeks ago this problem only appeared 2+ weeks ago and in brave browser views used to register immediately. By ignoring this you have created dia consequences to platform views. I for one am furious of the loss of that channel over this issue that i reported. If views go down due to a bug then they leave so what is the point of even running a platform? Users are obsessed with views and anything that compromises that then they will leave, obviously. The view system has gone to pot.

**Comment by DanielnetoDotCom** [comment]
> https://github.com/WWBN/AVideo/wiki/How-Video-Count-Works#checking-the-functionality

I am sorry for what happened

did you check this if something is wrong?

also did you check this? https://github.com/WWBN/AVideo/wiki/How-Video-Count-Works#why-is-my-video-view-not-added

**Comment by para2022** [comment]
Yes I have read them. No it does not take 30 seconds to register a view on other browsers or my iPhone. It works how it should. Brave takes 30 seconds to register and this is the problem. This problem never used to happen. Users don’t always hang around on a video for 30 seconds and the fact a user played a video regardless of how long is still a view.

**Comment by DanielnetoDotCom** [comment]
Thank you for clarifying. Just to confirm, you're saying that only on the Brave browser the view takes 30 seconds to register, while on other browsers and your iPhone, the view registers immediately? I want to ensure I fully understand the issue so I can address it appropriately.

**Comment by para2022** [comment]
Yes that is exactly it.

**Comment by DanielnetoDotCom** [comment]
I've recently conducted tests on the tutorials site using the Brave browser and everything seems to be functioning as expected on my end. One thing to note is that Brave is known to block ad scripts. It's possible that it's blocking a JavaScript ad script from your site, which in turn might be affecting the "addview" function. If that isn't the cause, I'm uncertain about the root of the issue. In that case, you'd need to delve into the JavaScript to pinpoint the reason.

BTW in the last update if the view is not added, the JS will show you the reason why

![image](https://github.com/WWBN/AVideo/assets/18009293/1697d43d-e74e-403a-9ec1-426706566b36)

**Comment by para2022** [comment]
Suddenly it is working. I open dev tools and it tells me if its already had a view from me if it has. If it hasnt recieved a view from me its registering it straight away now. Weird. 

Does this mean anything? 
default.css:1     Failed to load resource: the server responded with a status of 404 (Not Found)

**Comment by para2022** [comment]
ok well the views seem ok now. Il report back it if starts again.

**Comment by para2022** [comment]
Just had one register at 30 seconds and i had not viewed the video before so its appears to be an intermittent fault. I just cant get my head around it.

**Comment by para2022** [comment]
This video was just uploaded i was the first to click on it and you can see the marker at 30secs where the view registered and the dev tools say 'View not added, the user already have a view in this session'

![Screenshot 2023-08-21 at 15 46 33](https://github.com/WWBN/AVideo/assets/107706817/9d7e4ffd-8859-4440-ab84-b1b7084d8380)

**Comment by para2022** [comment]
it will have a view from me now. But it had zero views when it was just uploaded. In the screenshot it shows my registered view but only after 30 seconds where my mouse is. I viewed it and exactly at 30 seconds the view registered which is in the intial bug i reported. It did not happen on 6 others i tested so it is an intermittent bug.
